x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/action/service/TokenInfo.java

```diff
@@ -12,24 +12,34 @@
 import org.elasticsearch.common.io.stream.Writeable;
 import org.elasticsearch.common.xcontent.ToXContentObject;
 import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.core.Nullable;
 
 import java.io.IOException;
+import java.util.Collection;
 import java.util.Map;
 import java.util.Objects;
 
 public class TokenInfo implements Writeable, ToXContentObject, Comparable<TokenInfo> {
 
     private final String name;
     private final TokenSource source;
```

Comment:
The `source` seems weird now - file and index are handled so separately in the response objects, that it feels unnecessary to have this field.

Reply:
The source can be computed based on the new `nodeNames` field, which has value `null` for index-based tokens. So I removed this field, but kept the getter to compute it on the fly.